### PR TITLE
Handle reconnecting websocket cancel

### LINKED ISSUE #4858: [local-app] trying to reconnect to stopped supervisors forever
### Bug description

It seems to do with the local machine loosing connection and workspaces getting stopped in the meantime.

### Steps to reproduce

See above.

### Expected behavior

_No response_

### Example repository

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by rl-gitpod**
Fixes #4858

Pass in context to allow cancel to be handled and not stay in reconnection attempt loop.

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

Associated issue: *#4858*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/local-app/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/local-app/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by akosyakov — APPROVED** [review]
/lgtm

/hold 

@rl-gitpod Could you clean up commit to remove code which did not make it? then you can comment `/unhold` and the PR will be merged

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: e0df5b99dc8ae92948ae756774a0841990a26255</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5436#pullrequestreview-741406091" title="LGTM">akosyakov</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

Associated issue: *#4858*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- ~~[components/local-app/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/local-app/OWNERS)~~ [akosyakov]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by rl-gitpod** [comment]
@akosyakov - wasn't 100% sure what you were requesting but a squashed the commits & comments, which removed the lgtm

/unhold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: e0df5b99dc8ae92948ae756774a0841990a26255</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5436#issuecomment-908179998" title="LGTM">akosyakov</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

Associated issue: *#4858*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- ~~[components/local-app/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/local-app/OWNERS)~~ [akosyakov]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5436#issuecomment-908180367" title="Approved">akosyakov</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

Associated issue: *#4858*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- ~~[components/local-app/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/local-app/OWNERS)~~ [akosyakov]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5436#issuecomment-908180367" title="Approved">akosyakov</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5436#issuecomment-908935717" title="Approved">csweichel</a>*

Associated issue: *#4858*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [csweichel]
- ~~[components/local-app/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/local-app/OWNERS)~~ [akosyakov,csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
